### PR TITLE
Implementation of Sync Optimization for Movies, Series and Episodes

### PR DESCRIPTION
**Pull request by JaiZed**
Implemented algorithm, that (when the switches are on) will only check movies, series or episodes that are monitored, or whose monitored state does not agree between Bazarr and Radarr / Sonarr.
This can lead to a dramatic improvement in time spent synchronizing with Radarr and Sonarr if the user stops monitoring media that they are satisfied with in Radarr / Sonarr.

These settings are exposed in the UI under Settings -> Scheduler -> Sonarr/Radarr Sync.
- Sync Only Monitored Movies
- Sync Only Monitored Series
- Sync Only Monitored Episodes

### CODE REVIEW

File: bazarr/radarr/sync/movies.py [hunk 1]
```diff
@@ -45,6 +52,16 @@ def update_movie(updated_movie, send_event):
             event_stream(type='movie', action='update', payload=updated_movie['radarrId'])
 
 
+def get_movie_monitored_status(movie_id):
+    existing_movie_monitored = database.execute(
+        select(TableMovies.monitored)
+        .where(TableMovies.tmdbId == movie_id))\
+        .first()
+    if existing_movie_monitored == None:
```
Comment (morpheus65535) [inline bazarr/radarr/sync/movies.py:60]: `if not existing_movie_monitored:`

File: bazarr/radarr/sync/movies.py [hunk 2]
```diff
@@ -45,6 +52,16 @@ def update_movie(updated_movie, send_event):
...
+        return True
+    else:
+        return bool_map[existing_movie_monitored[0]]
```
Comment (morpheus65535) [inline bazarr/radarr/sync/movies.py:63]: `return bool_map[existing_movie_monitored['monitored']]`

File: bazarr/sonarr/sync/series.py [hunk 3]
```diff
@@ -16,6 +16,20 @@
 from .parser import seriesParser
 from .utils import get_profile_list, get_tags, get_series_from_sonarr_api
 
+# map between booleans and strings in DB
+bool_map = {"True": True, "False": False}
+
+FEATURE_PREFIX = "SYNC_SERIES "
+def trace(message):
+    if settings.general.debug:
+        logging.debug(FEATURE_PREFIX + message)
```
Comment (morpheus65535) [inline bazarr/sonarr/sync/series.py:25]: Any chance to not duplicate that part of code and have it once somewhere like bazarr/utilities/helper.py?

**Review by morpheus65535 — COMMENTED** [review]
Some code improvements (standardization) or comments. Thanks!

File: bazarr/sonarr/sync/series.py (hunk 3 above)
Comment (JaiZed) [reply bazarr/sonarr/sync/series.py:25]: I would have loved to do so, but I didn’t know where to put it.
I’ll put it there if that’s the best place.

Can I also create a constant for the magic number 20480?
That one has been bothering me.

File: bazarr/radarr/sync/movies.py (hunk 1 above)
Comment (JaiZed) [reply bazarr/radarr/sync/movies.py:60]: I’m not sure that is right.
I’m doing a bit of reading and they suggest:

` if existing_movie_monitored is None:`

File: bazarr/radarr/sync/movies.py (hunk 2 above)
Comment (JaiZed) [reply bazarr/radarr/sync/movies.py:63]: That doesn’t work.
I’m indexing into the tuple returned by the database call.

File: bazarr/radarr/sync/movies.py (hunk 1 above)
Comment (morpheus65535) [reply bazarr/radarr/sync/movies.py:60]: In fact, the sqlalchemy may return a single row or None. My code will check if it returned something in `[None, False, 0]`. `is None` is more precise but in our specific case it wont make a difference.

File: bazarr/radarr/sync/movies.py (hunk 2 above)
Comment (morpheus65535) [reply bazarr/radarr/sync/movies.py:63]: You're right, my bad. The result is still a tuple of a single row.

File: bazarr/sonarr/sync/series.py (hunk 3 above)
Comment (morpheus65535) [reply bazarr/sonarr/sync/series.py:25]: Forget code duplication, there's no real benefit in this specific case as you need to deal with context.

Yes adding a constant for 20480 in constants.py would be a good idea.

File: bazarr/radarr/sync/movies.py (hunk 1 above)
Comment (JaiZed) [reply bazarr/radarr/sync/movies.py:60]: If that’s true, that would change my logic in a bad way.
I want to return True only if the db record is missing.
If the db returns False, I want to return False, not True.
I will change it to use “is None”, though.

File: bazarr/sonarr/sync/series.py (hunk 3 above)
Comment (JaiZed) [reply bazarr/sonarr/sync/series.py:25]: Do you prefer lower case or all caps for constants?
e.g.

\# minimum file size for Bazarr to consider it a video
MINIMUM_VIDEO_SIZE = 20480

Comment (morpheus65535) [reply bazarr/sonarr/sync/series.py:25]: I've never did it that way but it seems to be the correct way of doing it. Lets go with uppercase and we'll fix the others in time.

Comment (JaiZed) [reply bazarr/sonarr/sync/series.py:25]: Done.

**Pull request merged**